docs/content/guides/developer/app-examples/e2e-counter.mdx

```diff
@@ -28,18 +28,47 @@ Before getting started, make sure you have:
 
 - [Installed the latest version of Sui](../getting-started/sui-install.mdx).
 
+## Directory structure
+
+To begin, create a new folder on your system titled `react-e2e-counter` to hold all your project files. You can name this directory differently, but the rest of the guide references this file structure. Inside that folder, create two more folders: `move` and `src`. Inside the `move` folder, create a `counter` directory. Finally, create a `sources` folder inside `counter`. Different projects have their own directory structure, but it's common to split code into functional groups to help with maintenance.
+
+:::checkpoint
+
+- You have the latest version of Sui installed. If you run `sui --version` in your terminal or console, it responds with the currently installed version.
+- Your active environment is pointing to the expected network. Run `sui client active-env` to make sure. If you receive a warning about a client and server API version mismatch, update Sui using the version in the relevant branch (`mainnet`, `testnet`, `devent`) of the Sui repo.
+- Your active address has SUI. Run `sui client balance` in your terminal or console. If there is no balance, [acquire SUI](../getting-started/get-coins.mdx) from the faucet (not available in Mainnet).
+- You have a directory to place the files you create in. The top-level directory is named `react-e2e-counter` if you want to match the directory structure in this guide. 
+
+:::
+
 ## Smart contracts {#smart-contracts}
 
 In this part of the guide, you write the Move contracts that create, increment, and reset counters. The first step is to [set up a Move package](../first-app/write-package.mdx) for storing your Move modules.
 
+### Move.toml
+
+To begin writing your smart contracts, create a `sources` folder in your `counter` folder (if using recommended directory names). Create a file inside the folder named `Move.toml` and copy the following code into it. This is the package manifest file. If you want to learn more about the structure of the file, see [Package Manifest](https://move-book.com/concepts/manifest.html) in The Move Book.
```

Comment:
These instructions seem to be duplicated from the directory structure setup. Should we avoid having them in both places?

Also if we have these instructions for manual directory setup, we may not need the sentence above: 

"The first step is to set up a Move package for storing your Move modules." 

This sentence points to another set of instructions for setting up the directory.

Reply:
Good call. I added the Directory Structure section to cover this info and forgot to update. And the other issue is the same issue with trustless swap. Corrected both.